docs/src/main/docs/cdi-extensions.adoc

```diff
@@ -0,0 +1,217 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon CDI extensions
+:description: Helidon CDI extensions guide
+:keywords: helidon, guide, CDI
+
+* <<Overview>>
+* <<Installing a CDI extension>>
+* <<Injecting an object using a CDI extension>>
+
+== Overview
+
+Helidon provides https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#spi[portable extensions] that you can use to inject the following objects into your Helidon MicroProfile applications:
+
+* HikariCP data sources
+* Jedis clients
+* Oracle Cloud Infrastructure Object Storage clients
+
+== Installing a CDI extension
+
+The Helidon CDI extensions are available as https://mvnrepository.com/artifact/io.helidon.integrations.cdi[Maven artifacts].
+
+To install an extension, ensure that the extension and its runtime dependencies are specified on your application's runtime classpath.
+
+The following examples show how to specify the required dependencies for the Helidon CDI extensions in your Maven POM (`pom.xml`).
+ 
+* To install the extension for HikariCP data sources, specify the following dependency:
+
+[source,xml]
+----
+<dependency>
+  <groupId>io.helidon.integrations.cdi</groupId>
+  <artifactId>helidon-integrations-cdi-datasource-hikaricp</artifactId>
+  <version>1.0.0</version>
+  <scope>runtime</scope>
+</dependency>
+----
+  
+* To install the extension for Jedis clients, specify the following dependency:
+ 
+[source,xml]
+----
+<dependency>
+  <groupId>io.helidon.integrations.cdi</groupId>
+  <artifactId>helidon-integrations-cdi-jedis</artifactId>
+  <version>1.0.0</version>
+  <scope>runtime</scope>
+</dependency>
+----
+  
+* To install the extension for Oracle Cloud Infrastructure Object Storage clients:
+
+   1. Clone the Oracle Cloud Infrastructure Java SDK artifacts. This step is necessary because, currently, the Oracle Cloud Infrastructure Java SDK is not available in the Maven repository.
```

Comment:
This step, and step 2.

Reply:
> step 1, here, is a way to make step 2 happen, and step 2 must be done because of OCI's absence from Maven Central.

I think that's implied, but there's no harm in stating it explicitly.

Reply:
I moved the 'why clone & install the jdk' info to the lead sentence.